indexer/ftypes_matcher.cpp

```diff
@@ -385,67 +386,67 @@ bool IsTypeConformed(uint32_t type, StringIL const & path)
   return true;
 }
 
-string DebugPrint(HighwayClass const cls)
+char const * HihgwayClassToString(HighwayClass const & cls)
 {
-  stringstream out;
-  out << "[ ";
   switch (cls)
   {
-    case HighwayClass::Undefined:
-      out << "Undefined";
-    case HighwayClass::Error:
-      out << "Error";
-    case HighwayClass::Trunk:
-      out << "Trunk";
-    case HighwayClass::Primary:
-      out << "Primary";
-    case HighwayClass::Secondary:
-      out << "Secondary";
-    case HighwayClass::Tertiary:
-      out << "Tertiary";
-    case HighwayClass::LivingStreet:
-      out << "LivingStreet";
-    case HighwayClass::Service:
-      out << "Service";
-    case HighwayClass::Count:
-      out << "Count";
-    default:
-      out << "Unknown value of HighwayClass: " << static_cast<int>(cls);
+    case HighwayClass::Undefined: return "Undefined";
+    case HighwayClass::Error: return "Error";
+    case HighwayClass::Trunk: return "Trunk";
+    case HighwayClass::Primary: return "Primary";
+    case HighwayClass::Secondary: return "Secondary";
+    case HighwayClass::Tertiary: return "Tertiary";
+    case HighwayClass::LivingStreet: return "LivingStreet";
+    case HighwayClass::Service: return "Service";
+    case HighwayClass::Pedestrian: return "Pedestrian";
+    case HighwayClass::Count: return "Count";
   }
-  out << " ]";
+}
+
+string DebugPrint(HighwayClass const cls)
+{
+  stringstream out;
+  out << "[ " << HihgwayClassToString(cls) << " ]";
   return out.str();
 }
 
 HighwayClass GetHighwayClass(feature::TypesHolder const & types)
 {
   Classificator const & c = classif();
-  static pair<HighwayClass, uint32_t> const kHighwayClasses[] = {
-      {HighwayClass::Trunk, c.GetTypeByPath({"highway", "motorway"})},
-      {HighwayClass::Trunk, c.GetTypeByPath({"highway", "motorway_link"})},
-      {HighwayClass::Trunk, c.GetTypeByPath({"highway", "trunk"})},
-      {HighwayClass::Trunk, c.GetTypeByPath({"highway", "trunk_link"})},
-      {HighwayClass::Trunk, c.GetTypeByPath({"route", "ferry"})},
-      {HighwayClass::Primary, c.GetTypeByPath({"highway", "primary"})},
-      {HighwayClass::Primary, c.GetTypeByPath({"highway", "primary_link"})},
-      {HighwayClass::Secondary, c.GetTypeByPath({"highway", "secondary"})},
-      {HighwayClass::Secondary, c.GetTypeByPath({"highway", "secondary_link"})},
-      {HighwayClass::Tertiary, c.GetTypeByPath({"highway", "tertiary"})},
-      {HighwayClass::Tertiary, c.GetTypeByPath({"highway", "tertiary_link"})},
-      {HighwayClass::LivingStreet, c.GetTypeByPath({"highway", "unclassified"})},
-      {HighwayClass::LivingStreet, c.GetTypeByPath({"highway", "residential"})},
-      {HighwayClass::LivingStreet, c.GetTypeByPath({"highway", "living_street"})},
-      {HighwayClass::Service, c.GetTypeByPath({"highway", "service"})},
-      {HighwayClass::Service, c.GetTypeByPath({"highway", "track"})}};
-  uint8_t const kTruncLevel = 2;
+
```

Comment:
I generally don't like the idea to load classifier on each GetHighwayClass() call and use it only once (during the first call). Why not to wrap whole map into individual class, which will fill m_highwayClasses map in ctor, and use it there?